inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/batch_to_space/batch_to_space_kernel_base.cpp

```diff
@@ -26,17 +26,32 @@ bool BatchToSpaceKernelBase::Validate(const Params& p, const optional_params& o)
         return false;
     }
 
+    const batch_to_space_params& params = static_cast<const batch_to_space_params&>(p);
+    for (auto& fused_op : params.fused_ops) {
+        if (!IsFusedPrimitiveSupported(fused_op))
+            return false;
+    }
+
+    if (params.inputs[0].Dimentions() > 6)
+        return false;
+
     return true;
 }
 
 CommonDispatchData BatchToSpaceKernelBase::SetDefault(const batch_to_space_params& params, const optional_params&) const {
-    CommonDispatchData runInfo;
-
-    std::vector<size_t> global = { params.output.Batch().v,
-                                   params.output.Feature().v,
-                                   params.output.W().v * params.output.Z().v * params.output.Y().v * params.output.X().v };
+    const auto& out = params.output;
 
-    auto local = GetOptimalLocalWorkGroupSizes(global, params.engineInfo);
+    CommonDispatchData runInfo;
+    std::vector<size_t> global;
+    std::vector<size_t> local;
+
+    if (out.GetLayout() == DataLayout::b_fs_yx_fsv16) {
+        global = { out.Batch().v, Align(out.Feature().v, 16), out.Y().v * out.X().v };
+        local = {1, 16, 1};
```

Comment:
As you align output feature number to 16, you should care about leftovers in your kernel. Or choose these workgroup sizes only if number of channels is multiple of 16